Branch-SDK/Branch-SDK/BNCContentDiscoveryManager.m

```diff
@@ -40,18 +40,14 @@ @implementation BNCContentDiscoveryManager
 
 - (NSString *)spotlightIdentifierFromActivity:(NSUserActivity *)userActivity {
 #if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 90000
-    if ([userActivity.activityType hasPrefix:BRANCH_SPOTLIGHT_PREFIX]) {
-        return userActivity.activityType;
-    }
+    // If it has our prefix, then the link identifier is just the last piece of the identifier.
+    NSString *activityIdentifier = userActivity.userInfo[CSSearchableItemActivityIdentifier];
+    BOOL isBranchIdentifier = [activityIdentifier hasPrefix:BRANCH_SPOTLIGHT_PREFIX];
     
-    // CoreSpotlight version. Matched if it has our prefix, then the link identifier is just the last piece of the identifier.
-    if ([userActivity.activityType isEqualToString:CSSearchableItemActionType]) {
-        NSString *activityIdentifier = userActivity.userInfo[CSSearchableItemActivityIdentifier];
-        BOOL isBranchIdentifier = [activityIdentifier hasPrefix:BRANCH_SPOTLIGHT_PREFIX];
-        
-        if (isBranchIdentifier) {
-            return activityIdentifier;
-        }
+    // Checking for CSSearchableItemActionType in the activity for legacy spotlight indexing (pre 0.12.7)
+    // Now we index NSUserActivies with type set to io.branch. + bundleId for better SEO
+    if ([userActivity.activityType isEqualToString:CSSearchableItemActionType] || isBranchIdentifier) {
+        return activityIdentifier;
```

Comment:
@aaustin here is the fallback for existing integrations. The activity type is always set to `com.apple.corespotlightitem` which is the value that maps to this macro `CSSearchableItemActionType`.  This is the spotlight indexing activity, w/e we did set to `NSUserActivity` never made it to Spotlight. Let me know if I am missing anything.